pkg/diff/diff.go

```diff
@@ -97,6 +98,42 @@ func DiffUnstructureds(x, y unstructured.Unstructured, key string, opts ...Optio
 	return d.result, nil
 }
 
+// DiffStructs calulates the diff between two struct objects,
+// which are not k8s manifests.
+func DiffStructs(x, y interface{}, key string, opts ...Option) (*Result, error) {
+	map_x := map[string]interface{}{}
+	map_y := map[string]interface{}{}
+	yml_x, err := yaml.Marshal(x)
+	if err != nil {
+		return nil, err
+	}
+	yml_y, err := yaml.Marshal(y)
+	if err != nil {
+		return nil, err
+	}
+	yaml.Unmarshal(yml_x, &map_x)
+	yaml.Unmarshal(yml_y, &map_y)
+
+	var (
+		path = []PathStep{}
+		vx   = reflect.ValueOf(map_x)
+		vy   = reflect.ValueOf(map_y)
+		d    = &differ{result: &Result{}}
+	)
+	for _, opt := range opts {
+		opt(d)
+	}
+
+	d.initIgnoredPaths(key)
```

Comment:
Not sure this would work? 🤔 Or could you introduce the possible key for this? Since this key in case of k8s manifests could look like this https://pipecd.dev/docs-v0.47.x/user-guide/managing-application/configuration-drift-detection/#ignore-drift-detection-for-specific-fields